cmd/juju/machine/upgradeseries_test.go

```diff
@@ -144,7 +109,23 @@ func (s *UpgradeSeriesSuite) TestCompleteCommandDoesNotAcceptSeries(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, "wrong number of arguments")
 }
 
-func (s *UpgradeSeriesSuite) TestPrepareCommandShouldAcceptAgree(c *gc.C) {
-	err := s.runUpgradeSeriesCommandWithoutConfirmation(c, machine.PrepareCommand, machineArg, seriesArg, "--agree")
+func (s *UpgradeSeriesSuite) TestPrepareCommandShouldPromptUserForConfirmation(c *gc.C) {
+	err := s.runUpgradeSeriesCommandWithConfirmation(c, "y", machine.PrepareCommand, machineArg, seriesArg)
 	c.Assert(err, jc.ErrorIsNil)
+	confirmationMsg := fmt.Sprintf(machine.UpgradeSeriesConfirmationMsg, machineArg, seriesArg)
+	c.Assert(s.ctx.Stdout.(*bytes.Buffer).String(), gc.Equals, confirmationMsg)
+}
+
+func (s *UpgradeSeriesSuite) TestPrepareCommandShouldAcceptAgreeAndNotPrompt(c *gc.C) {
+	err := s.runUpgradeSeriesCommandWithConfirmation(c, "n", machine.PrepareCommand, machineArg, seriesArg, "--agree")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(s.ctx.Stdout.(*bytes.Buffer).String(), gc.Equals, ``)
+}
+
+func stubRemoteAPI(c *gc.C) *mocks.MockUpgradeMachineSeriesAPI {
+	mockController := gomock.NewController(c)
```

Comment:
Even though this is for a stub and the only expectations use ```AnyTimes()```, I would still call ```Finish()``` on the controller at some point to ensure you did not get calls had no accompanying expectation (I *think* unexpected methods with no return would not break tests here).

I generally either return the controller from, or pass one into such methods.

Reply:
I don't agree